src/libraries/Microsoft.Extensions.Caching.Abstractions/src/PostEvictionCallbackRegistration.cs

```diff
@@ -5,8 +5,8 @@ namespace Microsoft.Extensions.Caching.Memory
 {
     public class PostEvictionCallbackRegistration
     {
-        public PostEvictionDelegate EvictionCallback { get; set; }
+        public PostEvictionDelegate EvictionCallback { get; set; } = null!;
```

Comment:
This is a lie. The property should be annotated as nullable.

```suggestion
        public PostEvictionDelegate? EvictionCallback { get; set; }
```

Reply:
`CacheEntryExtensions.RegisterPostEvictionCallback` would throw on null `EvictionCallback`
Called here: https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Caching.Abstractions/src/CacheEntryExtensions.cs#L189

Reply:
This class has a default ctor. So just doing:

```C#
new PostEvictionCallbackRegistration()
```

creates an object with a null EvictionCallback.

> CacheEntryExtensions.RegisterPostEvictionCallback would throw on null EvictionCallback

That's a bug.

Reply:
Created an issue: https://github.com/dotnet/runtime/issues/64147